huaweicloud/services/fgs/resource_huaweicloud_fgs_trigger.go

```diff
@@ -492,6 +674,37 @@ func setKafkaEventData(d *schema.ResourceData, eventData map[string]interface{})
 	return d.Set("kafka", result)
 }
 
+func setSharedApigEventData(d *schema.ResourceData, eventData map[string]interface{}) error {
+	funcInfo := eventData["func_info"].(map[string]interface{})
+	result := []map[string]interface{}{
+		{
+			"group_id":                eventData["group_id"],
+			"api_name":                eventData["api_name"],
+			"env_name":                eventData["env_name"],
+			"security_authentication": eventData["auth"],
+			"request_protocol":        eventData["protocol"],
+			"timeout":                 funcInfo["timeout"],
+		},
+	}
+	return d.Set("apig", result)
+}
+
+func setDedicatedApigEventData(d *schema.ResourceData, eventData map[string]interface{}) error {
```

Comment:
the repetition with `setSharedApigEventData` is too high, and it is recommended to combine them